mongo/integration/client_test.go

```diff
@@ -515,7 +515,8 @@ func TestClient(t *testing.T) {
 		assert.Nil(t, err, "unexpected error calling Ping: %v", err)
 	})
 
-	mt.Run("minimum RTT is monitored", func(mt *mtest.T) {
+	rtt90Opts := mtest.NewOptions().MinServerVersion("4.4")
```

Comment:
Before this change, we always started the RTT monitor. The goals of this ticket is to avoid opening a connection for streaming for servers that do not support streaming: https://github.com/mongodb/specifications/pull/1452/files#diff-67f5deaee17cfd34b1845819d83009a9b7e16d2ef28ffce946f51b714474efdeR745

Reply:
If I'm reading the specification code correctly, and RTT monitor will not start unless a server supports streaming. A server will only support streaming if handshake responses include the topologyVersion, which will only occur in versions >= 4.4. Therefore, IIUC there is no need to run RTT tests on versions < 4.4.

Reply:
Was the RTT monitor intended to be disabled for server versions earlier than 4.4 before this change? If not, that seems like a conflation between streaming disabled for FaaS and streaming not supported. I asked about it on the spec PR [here](https://github.com/mongodb/specifications/pull/1452/files#r1334705593).

Reply:
I think the Go Driver was incorrectly implementing these tests. AFAICT RTT was added specifically to support streaming ([see SPEC-1579](https://jira.mongodb.org/browse/SPEC-1579)), and from the SDAM specifications on the Polling Protocol:

> The polling protocol is used to monitor MongoDB <= 4.4 servers.

Reply:
Makes sense based on how the concept of an RTT monitor was added. However, the spec does specifically call out that drivers can use the RTT monitor for any server version in [Measuring RTT](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#measuring-rtt):
> For consistency, clients MAY use dedicated connections to measure RTT for all servers, even those that do not support awaitable hello or legacy hello.

The updated pseudo-code is _technically_ compliant with the spec, but suggests an additional requirement that is new for some drivers (e.g. the Go driver).

Reply:
I haven't heard back from anyone on the spec PR, but I think we should avoid changing unrelated behavior unless it's really obvious that is the intent. Disabling the RTT monitor when streaming heartbeats aren't supported effectively disables the RTT-short-circuit feature of CSOT for server version 3.6 - 4.2, which is not related to improving behavior in FaaS environments.

Reply:
@matthewdale The specifications PR has been updated to to say clients MUST NOT dedicated connections to measure RTT in this case.

Reply:
Agreed. However, we should still expect RTT to be monitored on all supported server versions. We should keep this test the same (i.e. don't limit to 4.4+) and make the necessary changes to record RTT times when using polling monitoring.